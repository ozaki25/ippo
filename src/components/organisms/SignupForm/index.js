import React from 'react';
import { Button, TextField, Typography } from '@material-ui/core';
import propTypes from 'prop-types';

class SignupForm extends React.Component {
  state = { email: '', pass: '', name: '', loading: false, error: '' };

  onClick = async event => {
    event.preventDefault();
    const { email, pass, name } = this.state;
    this.setState({ loading: true });
    try {
      await this.props.onSubmit({ data: { email, pass, name } });
    } catch (e) {
      this.setState({ error: e.toString() });
    } finally {
      this.setState({ loading: false });
    }
  };

  onChange = event => this.setState({ [event.target.name]: event.target.value });

  render() {
    const { email, pass, name, loading, error } = this.state;
    const invalid = email.trim() === '' || pass.trim() === '' || name.trim() === '';
    return (
      <form>
        {error && <Typography color="error">{error}</Typography>}
        <TextField
          label="名前"
          name="name"
          value={this.state.name}
          onChange={this.onChange}
          margin="normal"
          color="primary"
          fullWidth
          required
        />
        <br />
        <TextField
          label="メールアドレス"
          name="email"
          value={this.state.email}
          onChange={this.onChange}
          margin="normal"
          color="primary"
          fullWidth
          required
        />
        <br />
        <TextField
          label="パスワード"
          name="pass"
          type="password"
          value={this.state.pass}
          onChange={this.onChange}
          margin="normal"
          color="primary"
          fullWidth
          required
        />
        <br />
        <Button
          onClick={this.onClick}
          type="submit"
          disabled={invalid || loading}
          color="primary"
          variant="contained"
          fullWidth
        >
          登録
        </Button>
      </form>
    );
  }
}

SignupForm.displayName = 'SignupForm';

SignupForm.propTypes = {
  onSubmit: propTypes.func.isRequired,
};

export default SignupForm;
