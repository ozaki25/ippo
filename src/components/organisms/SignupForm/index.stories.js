import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import SignupForm from '.';

const stories = storiesOf('organisms/SignupForm', module);

stories.add('通常パターン', () => <SignupForm onSubmit={action('onSubmit')} />);

stories.add('エラー', () => (
  <SignupForm
    onSubmit={() =>
      new Promise((resolve, reject) =>
        reject(new Error('The email address is already in use by another account.')),
      )
    }
  />
));
