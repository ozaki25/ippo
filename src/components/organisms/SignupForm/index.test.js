import React from 'react';
import { snapshot } from 'src/test/helpers';
import SignupForm from '.';

const props = {
  onSubmit: jest.fn(),
};

snapshot('SignupForm/nomal', <SignupForm {...props} />);
