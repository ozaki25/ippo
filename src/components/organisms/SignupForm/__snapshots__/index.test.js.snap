// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: SignupForm/nomal 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <SignupForm
    onSubmit={[MockFunction]}
  >
    <form>
      <Blueprint3.FormGroup
        label="名前"
        labelFor="name"
      >
        <div
          className="bp3-form-group"
        >
          <label
            className="bp3-label"
            htmlFor="name"
          >
            名前
             
            <span
              className="bp3-text-muted"
            />
          </label>
          <div
            className="bp3-form-content"
          >
            <Blueprint3.InputGroup
              id="name"
              large={true}
              name="name"
              onChange={[Function]}
              value=""
            >
              <div
                className="bp3-input-group bp3-large"
              >
                <Blueprint3.Icon />
                <input
                  className="bp3-input"
                  id="name"
                  name="name"
                  onChange={[Function]}
                  style={
                    Object {
                      "paddingRight": 10,
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </Blueprint3.InputGroup>
          </div>
        </div>
      </Blueprint3.FormGroup>
      <Blueprint3.FormGroup
        label="メールアドレス"
        labelFor="email"
      >
        <div
          className="bp3-form-group"
        >
          <label
            className="bp3-label"
            htmlFor="email"
          >
            メールアドレス
             
            <span
              className="bp3-text-muted"
            />
          </label>
          <div
            className="bp3-form-content"
          >
            <Blueprint3.InputGroup
              id="email"
              large={true}
              name="email"
              onChange={[Function]}
              value=""
            >
              <div
                className="bp3-input-group bp3-large"
              >
                <Blueprint3.Icon />
                <input
                  className="bp3-input"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  style={
                    Object {
                      "paddingRight": 10,
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </Blueprint3.InputGroup>
          </div>
        </div>
      </Blueprint3.FormGroup>
      <Blueprint3.FormGroup
        label="パスワード"
        labelFor="pass"
      >
        <div
          className="bp3-form-group"
        >
          <label
            className="bp3-label"
            htmlFor="pass"
          >
            パスワード
             
            <span
              className="bp3-text-muted"
            />
          </label>
          <div
            className="bp3-form-content"
          >
            <Blueprint3.InputGroup
              id="pass"
              large={true}
              name="pass"
              onChange={[Function]}
              type="password"
              value=""
            >
              <div
                className="bp3-input-group bp3-large"
              >
                <Blueprint3.Icon />
                <input
                  className="bp3-input"
                  id="pass"
                  name="pass"
                  onChange={[Function]}
                  style={
                    Object {
                      "paddingRight": 10,
                    }
                  }
                  type="password"
                  value=""
                />
              </div>
            </Blueprint3.InputGroup>
          </div>
        </div>
      </Blueprint3.FormGroup>
      <Blueprint3.Button
        disabled={true}
        fill={true}
        large={true}
        onClick={[Function]}
        text="登録"
        type="submit"
      >
        <button
          className="bp3-button bp3-disabled bp3-fill bp3-large"
          disabled={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          tabIndex={-1}
          type="submit"
        >
          <Blueprint3.Icon
            key="leftIcon"
          />
          <span
            className="bp3-button-text"
            key="text"
          >
            登録
          </span>
          <Blueprint3.Icon
            key="rightIcon"
          />
        </button>
      </Blueprint3.Button>
    </form>
  </SignupForm>
</MockRouter>
`;
